src/utils.ts

```diff
@@ -70,3 +70,9 @@ export async function getBlockNum(network) {
 
   return blockNum;
 }
+
+export function checkSnapshot(snapshot, network) {
+  return (
+    snapshot !== 'latest' || !blockNumByNetwork[network] || snapshot > blockNumByNetwork[network]
```

Comment:
Function name is too generic, and did you tested your change? This `snapshot !== 'latest'` will be true all the time, wont work as expected.